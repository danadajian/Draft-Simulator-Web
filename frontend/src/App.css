html, body {
    height: 100%;
}

html {
    display: table;
    margin: auto;
}

body {
  display: table-cell;
  vertical-align: middle;
  background-color: darkslategray;
}

.App-header {
  text-align: center;
  color: navajowhite;
}

.Info-buttons {
  display: flex;
  flex-direction: column;
}

.Player-list-box-div {
  display: flex;
  float: left;
  margin: 1%;
}

.Player-list-box {
  background-color: lightblue;
}

.Player-button {;
  display: flex;
  flex-direction: column;
  margin: 1%;
  float: left;
}

.Add-button {;
  width: 80px;
  background-color: green;
}

.Remove-button {;
  width: 80px;
  background-color: red;
}

.Clear-button {;
  width: 80px;
  background-color: yellow;
}

.User-list-box-div {
  display: flex;
  float: left;
  margin: 1%;
}

.User-list-box {
  background-color: lightblue;
}

.Draft-button-div {;
  float: left;
  margin: 1%;
}

.Draft-button {;
  width: 80px;
  background-color: orange;
}

.Draft-results-div {;
  display: flex;
  float: left;
  margin: 1%;
}

.Slider-div {
  display: flex;
  flex-direction: column;
  float: left;
  margin-left: 2%;
  text-align: center;
  color: navajowhite;
}

.Loading-text {
  font-size: x-large;
  font-weight: bold;
  font-style: oblique;
  text-align: center;
  color: navajowhite;
  margin-bottom: 30%;
}

.App-logo {
  animation: App-logo-spin infinite 5s linear;
  height: 30vmin;
  pointer-events: none;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
