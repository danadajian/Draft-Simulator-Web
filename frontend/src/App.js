import React from 'react';
import './App.css';
import PlayerListBox from './PlayerListBox';
import $ from 'jquery';

class App extends React.Component {

    constructor(props) {
        super(props);
        this.state = {players: [], userPlayers: [], isLoading: true};
    }

    componentDidMount() {
        this.setState({isLoading: true});

        $.get(window.location.href + 'players', (data) => {
            const playerList = data.substring(2, data.length - 2).split(/', '|", '|', "/);
            this.setState({players: playerList, isLoading: false});
        });
    }

    addPlayers = () => {
        let currentPlayers = this.state.userPlayers;
        let selectedItems = this.refs.playerListbox.getSelectedItems();
        let playersToAdd = [];

        for (let i = 0; i < selectedItems.length; i++) {
            if (!currentPlayers.includes(selectedItems[i].label)) {
                playersToAdd.push(selectedItems[i].label);
            }
        }

        for (let i = 0; i < playersToAdd.length; i++) {
            this.refs.userListbox.addItem(playersToAdd[i]);
        }

        let totalPlayers = currentPlayers + playersToAdd;
        this.setState({userPlayers: totalPlayers});

        this.refs.playerListbox.clearSelection();
        this.refs.userListbox.clearSelection();
    };

    removePlayers = () => {
        let currentPlayers = this.state.userPlayers;
        let selectedUserItems = this.refs.userListbox.getSelectedItems();
        let removedPlayers = [];

        if (selectedUserItems.length > 0) {
            for (let i = 0; i < selectedUserItems.length; i++) {
                this.refs.userListbox.removeItem(selectedUserItems[i].label);
                removedPlayers.push(selectedUserItems[i].label);
            }
        }

        let remainingPlayers = [];
        for (let i = 0; i < currentPlayers.length; i++) {
            if (!removedPlayers.includes(currentPlayers[i])) {
                remainingPlayers.push(currentPlayers[i]);
            }
        }

        this.setState({userPlayers: remainingPlayers});

        this.refs.userListbox.clearSelection();
    };

    clearPlayers = () => {
        this.refs.userListbox.clear();
        this.setState({userPlayers: []});
        this.refs.userListbox.clearSelection();
    };

    render() {
        const {players, userPlayers, isLoading} = this.state;

        if (isLoading) {
            return <p>Loading players from ESPN . . .</p>;
        }

        return (
            <div>
                <PlayerListBox ref='playerListbox'
                               width={250} height={300}
                               source={players} multiple={true}
                               multipleextended={true}
                               className={"Player-list-box"}
                />
                <button onClick={this.addPlayers} style={{fontSize: 16}} className={"Add-button"}>Add</button>
                <button onClick={this.removePlayers} style={{fontSize: 16}} className={"Remove-button"}>Remove</button>
                <button onClick={this.clearPlayers} style={{fontSize: 16}} className={"Clear-button"}>Clear</button>
                <PlayerListBox ref='userListbox'
                             width={250} height={300}
                             source={userPlayers} multiple={true} multipleextended={true}
                             allowDrag={true} allowDrop={true}
                             className={"User-list-box"}
                />
            </div>
        )
    }
}

export default App;

// <img src={logo} className="App-logo" alt="logo"/>