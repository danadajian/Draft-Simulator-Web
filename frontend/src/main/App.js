import React, { Component } from 'react';
import './App.css';
import { Homepage } from "./Homepage";
import { Optimizer } from './optimizer/Optimizer'
import { Simulator } from "./simulator/Simulator";

class App extends Component {

    render() {
        if (window.location.pathname === '/home') {
            return <Homepage/>
        } else if (window.location.pathname === '/simulate') {
            return <Simulator/>
        } else if (window.location.pathname === '/optimize') {
            return <Optimizer/>
        }
    }
}

export default App;
