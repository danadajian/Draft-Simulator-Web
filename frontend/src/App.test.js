import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, mount } from 'enzyme';
import App from './App';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});


test('there are players', () => {
    const wrapper = shallow(<App players={[]}/>);
    expect(wrapper).toMatchSnapshot();
});
