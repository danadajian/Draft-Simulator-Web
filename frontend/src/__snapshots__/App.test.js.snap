// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`there are players 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    players={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Loading players from ESPN . . .",
    },
    "ref": null,
    "rendered": "Loading players from ESPN . . .",
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading players from ESPN . . .",
      },
      "ref": null,
      "rendered": "Loading players from ESPN . . .",
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
